crates/dbsp/src/trace/consolidation/tests/proptests.rs

```diff
@@ -236,4 +236,17 @@ proptest! {
         prop_assert_eq!(values2, expected_values2);
         prop_assert_eq!(values3, expected_values3);
     }
+
+    #[test]
+    fn quicksort_correctness(mut batch in vec(any::<(u16, ())>(), 0..50_000)) {
```

Comment:
This test wouldn't catch the bug addressed by this PR, which would require a non-unit value type.

Reply:
It'll still detect missed/duplicate values, with non-zst values sorting stability comes into play